gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Dataset.java

```diff
@@ -235,4 +219,39 @@ public Table create(String table, TableDefinition definition, BigQuery.TableOpti
   public BigQuery bigquery() {
     return bigquery;
   }
+
+  static Builder builder(BigQuery bigquery, DatasetId datasetId) {
+    return new Builder(bigquery).datasetId(datasetId);
```

Comment:
My comment above is about the Builder's constructors and not about the static builder methods.
As for the static builder methods of functional objects, I do think they should not be public, however
I don't mind if we also keep them as package-scope as long as they are used by our non-test code (which does
not seem to be the case here, and therefore should be removed).